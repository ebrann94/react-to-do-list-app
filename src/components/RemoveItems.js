import React from 'react';

const RemoveItems = (props) => (
    <div className="remove-items">
        <button className="remove-all-btn" onClick={props.handleRemoveAll} >Remove All</button>
        <button className="remove-completed-btn" onClick={props.handleRemoveCompleted} >Remove Completed</button>
    </div>
);

export default RemoveItems;