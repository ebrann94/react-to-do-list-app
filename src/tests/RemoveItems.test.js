import React from 'react';
import { shallow } from 'enzyme';

import RemoveItems from '../components/RemoveItems';
import { exportAllDeclaration } from '@babel/types';
import { findNodeModule } from 'jest-resolve';

describe('Remove Items components', () => {
    it('renders with out crashing', () => {
        const wrapper = shallow(<RemoveItems />);
        expect(wrapper).toMatchSnapshot();
    });

    it('calls removeAll function correctly', () => {
        const props = {
            handleRemoveAll: jest.fn(),
            handleRemoveCompleted: jest.fn()
        }

        const wrapper = shallow(<RemoveItems {...props}/>);
        wrapper.find('.remove-all-btn').simulate('click');
        expect(props.handleRemoveAll).toHaveBeenCalled();
        wrapper.find('.remove-completed-btn').simulate('click');
        expect(props.handleRemoveCompleted).toHaveBeenCalled();
    })
})