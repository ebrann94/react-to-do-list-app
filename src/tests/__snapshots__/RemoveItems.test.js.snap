// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remove Items components renders with out crashing 1`] = `ShallowWrapper {}`;
